@here/harp-mapview/lib/Tile.ts

```diff
@@ -517,12 +521,16 @@ export class Tile implements CachedResource {
      */
     removeUserTextElement(textElement: TextElement): boolean {
         const foundIndex = this.m_userTextElements.elements.indexOf(textElement);
-        if (foundIndex >= 0) {
-            this.m_userTextElements.elements.splice(foundIndex, 1);
-            this.textElementsChanged = true;
-            return true;
+        if (foundIndex === -1) {
+            return false;
         }
-        return false;
+
+        if (this.m_textElementsChanged === false) {
+            this.m_userTextElements = this.m_userTextElements.clone();
```

Comment:
same here, clone the group when it's changed.

Reply:
> same here, clone the group when it's changed.

This is cloning if the textElements **didnt** change, is that wanted?

Reply:
Yes, we want to clone the text element group once, just when we are about to change it since the last time it was rendered. 
After the first change (m_textElementsChanged === true), we don't need to clone it anymore.